pkg/option/config.go

```diff
@@ -2143,6 +2144,90 @@ func (c *DaemonConfig) Populate() {
 		c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
 	}
 
+	// BPF map size options
+	// Any map size explicitly set via option will override the dynamic
+	// sizing.
+	c.CTMapEntriesGlobalTCP = viper.GetInt(CTMapEntriesGlobalTCPName)
+	c.CTMapEntriesGlobalAny = viper.GetInt(CTMapEntriesGlobalAnyName)
+	c.NATMapEntriesGlobal = viper.GetInt(NATMapEntriesGlobalName)
+	c.PolicyMapMaxEntries = viper.GetInt(PolicyMapEntriesName)
+
+	if viper.GetBool(MapEntriesGlobalDynamicSizeName) {
```

Comment:
Can you refactor this into a calculation function so we can unit test the calculation?